tests/functional/selected_resources/test_selected_resources.py

```diff
@@ -0,0 +1,81 @@
+import pytest
+from dbt.tests.util import run_dbt
+from tests.functional.selected_resources.fixtures import (
+    on_run_start_macro_assert_selected_models_expected_list,
+    my_model1,
+    my_model2,
+    my_snapshot,
+)
+
+
+@pytest.fixture(scope="class")
+def macros():
+    return {
+        "assert_selected_models_expected_list.sql": on_run_start_macro_assert_selected_models_expected_list,
+    }
+
+
+@pytest.fixture(scope="class")
+def models():
+    return {"model1.sql": my_model1, "model2.sql": my_model2}
+
+
+@pytest.fixture(scope="class")
+def snapshots():
+    return {
+        "my_snapshot.sql": my_snapshot,
+    }
+
+
+@pytest.fixture(scope="class")
+def project_config_update():
+    return {
+        "on-run-start": "{{ assert_selected_models_expected_list(var('expected_list',[])) }}",
+    }
+
+
+def test_selected_resources_build(project):
+    results = run_dbt(
+        [
+            "build",
+            "--select",
+            "model1+",
```

Comment:
Should we do a `model2+` as one of the test cases? since the result of this selection is the same as the no select specified test below.

Reply:
I just modified a test to be `model2+`

It was failing at first (because `model1` was not built) so I moved all the tests to a Class that uses the fixture `build_all`, meaning that a full `dbt build` is run before each test (which allows us to select nodes that are depending on other ones)